### PR TITLE
filter testplans with metadata added from testcases

### PR DESCRIPTION
**Pull request by criley-eagle**
When filtering test plans, include the meta data from all test cases to see if the test plan should be included in classes to at least try to run.

### CODE REVIEW

File: testah-junit/src/test/java/org/testah/framework/cli/TestFilterPlatformTest.java [hunk 1]
```diff
@@ -68,24 +68,24 @@ private void testFilterMyPlatform(final int expectedTest1, final int expectedTes
         Set<Class<?>> classes = ImmutableSet.copyOf(Arrays.asList(classesToAdd));
 
         TS.params().setFilterByPlatform(null);
-        Assert.assertEquals(expectedTest1, filter.resetTestClassesMetFilters().filterTestPlansToRun(classes)
+        Assert.assertEquals("Test 1", expectedTest1, filter.resetTestClassesMetFilters().filterTestPlansToRun(classes)
                 .size());
 
         TS.params().setFilterByPlatform("");
```
Comment (gweiskeeagle) [inline testah-junit/src/test/java/org/testah/framework/cli/TestFilterPlatformTest.java:74]: no spaced after comma, line 75, 79, 83

File: testah-junit/src/main/java/org/testah/framework/testPlan/Junit5TestPlanSupport.java [hunk 2]
```diff
@@ -1,13 +1,6 @@
...
 
-import org.junit.jupiter.api.extension.AfterAllCallback;
-import org.junit.jupiter.api.extension.AfterEachCallback;
-import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
-import org.junit.jupiter.api.extension.BeforeAllCallback;
-import org.junit.jupiter.api.extension.BeforeEachCallback;
-import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
-import org.junit.jupiter.api.extension.ExtensionContext;
-import org.junit.jupiter.api.extension.TestWatcher;
+import org.junit.jupiter.api.extension.*;
```
Comment (gweiske) [inline testah-junit/src/main/java/org/testah/framework/testPlan/Junit5TestPlanSupport.java:3]: wildcard import?

File: testah-junit/src/main/java/org/testah/framework/dto/TestPlanAnnotationDto.java [hunk 3]
```diff
@@ -169,4 +180,52 @@ public String owner() {
         return owner;
     }
 
+    /**
+     * Merge testcase meta data into the testplan for filtering. JUnit4.
+     * @return returns the merged testplan meta.
+     */
+    public TestPlanAnnotationDto mergeWithTestCases() {
```
Comment (gweiske) [inline testah-junit/src/main/java/org/testah/framework/dto/TestPlanAnnotationDto.java:187]: i am not sure about merging the all test case runTypes for a TestPlan together. Suppose a test plan has 3 test cases. The test plan has no annotation but test case 1 has NOT_DEV, test case 2, has NOT_STAGE, test case 3 has NOT_PROD. The test plan ends up with NOT_DEV, NOT_STAGE, NOT_PROD.

Comment (criley-eagle) [reply testah-junit/src/main/java/org/testah/framework/dto/TestPlanAnnotationDto.java:187]: redesign

**Pull request closed without merging**